test/e2e/olm_test.go

```diff
@@ -0,0 +1,219 @@
+func testGuestClusterCatalogReady(parentCtx context.Context, client crclient.Client) func(t *testing.T) {
+	return func(t *testing.T) {
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(parentCtx)
+		defer cancel()
+
+		guestClusterCatalogSource := &operatorsv1alpha1.CatalogSource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      guestClusterCatalogSourceName,
+				Namespace: openshiftMarketplaceNamespace,
+			},
+		}
+
+		// The guest cluster catalogSource should eventually become available
+		err := wait.PollImmediateUntil(5*time.Second, func() (bool, error) {
+			if err := client.Get(ctx, crclient.ObjectKeyFromObject(guestClusterCatalogSource), guestClusterCatalogSource); err != nil {
+				t.Logf("failed to get guestcluster catalogsource %s/%s: %s", guestClusterCatalogSource.Namespace, guestClusterCatalogSource.Name, err)
+				return false, nil
+			}
+			if guestClusterCatalogSource.Status.GRPCConnectionState == nil {
+				return false, nil
+			}
+
+			return guestClusterCatalogSource.Status.GRPCConnectionState.LastObservedState == "READY", nil
+		}, ctx.Done())
+		g.Expect(err).NotTo(HaveOccurred(), "guestCluster catalogSource never became READY")
+		t.Logf("guestCluster catalogSource became available")
+	}
+}
+
+// testOperatorInstallationFromSource ensures that an operator can be installed from a catalogSource in the openshift-marketplace namespace.
+func testOperatorInstallationFromSource(parentCtx context.Context, client crclient.Client, catalogSourceName string) func(t *testing.T) {
+	return func(t *testing.T) {
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(parentCtx)
+		defer cancel()
+
+		testNamespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "olm-test-",
+			},
+		}
+		err := client.Create(ctx, testNamespace)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to create guest cluster namespace")
+		defer func() {
+			err = client.Delete(ctx, testNamespace)
+			g.Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("failed to delete test namespace %s", testNamespace.Name))
+		}()
+		t.Logf("Created namespace %s for test", testNamespace.Name)
+
+		operatorGroup := &operatorsv1.OperatorGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: testNamespace.Name,
+				Name:      "test-og",
+			},
+			Spec: operatorsv1.OperatorGroupSpec{
+				TargetNamespaces: []string{testNamespace.Name},
+			},
+		}
+		err = client.Create(ctx, operatorGroup)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to create guest cluster operatorGroup")
+		t.Logf("Created opeatorGroup %s/%s for test", operatorGroup.Namespace, operatorGroup.Name)
+
+		subscription := &operatorsv1alpha1.Subscription{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: testNamespace.Name,
+				Name:      "test-sub",
+			},
+			Spec: &operatorsv1alpha1.SubscriptionSpec{
+				InstallPlanApproval:    "Automatic",
+				Channel:                "stable",
+				Package:                "local-storage-operator",
```

Comment:
Ideally we would not rely on a third party operator being available in the Red Hat operators catalogSource. If we submit the [combo-operator](https://github.com/operator-framework/combo) to the community-operators catalog, we should rely on it instead.

Reply:
Worth adding this note as a TODO in the code?